roles/foreman_provisioning/tasks/configure_centos_9.yml

```
@@ -0,0 +1,40 @@
+- name: "set CentOS medium name"
+  set_fact:
+    centos_medium_name: "{{ 'CentOS Stream 9 mirror' if foreman_provisioning_foreman_version == 'nightly' or (foreman_provisioning_foreman_version is version_compare('2.1', '>=')) else 'CentOS mirror' }}"
```

No need to think about Foreman 2.0 and older anymore
```suggestion
    centos_medium_name: "CentOS Stream 9 mirror"
```